### PR TITLE
Add fitters

### LINKED ISSUE #41: Rethink dtypes
We need to keep the dtypes of tensors/arrays consistent across the entire modelling workflow. There are at least two ways we could do this:

1. Use a global/environment variable that sets the dtype. This can however, lead to a confusion and silent errors, see https://github.com/numpy/numpy/issues/6860
2. Add a `dtype` attribute to each relevant class, function that explicitly sets the dtype of relevant tensors/arrays. This is better than the first option because dtypes are set explicitly.

### PR DESCRIPTION
**Pull request by maltelueken**
Closes #41 

Adds multiple parameter fitters:

- `SGDFitter`: Parameter fitting with stochastic gradient descent. Iteratively updates parameters for all data batches (e.g., voxels) simultaneously. Returns the final parameters and the average loss across all data batches over the iterations.
- `GridFitter`: Parameter fitting with grid search. Creates a grid of parameter combinations and iteratively evaluates model predictions for chunks of the grid. This chunking is done to prevent memory problems as the grid can become large very quickly. Returns the final parameters and the minimal loss for each data batch.
- `LeastSquaresFitter`: Parameter fitting with least squares. Updates one or two parameters (intercept and slope) by minimizing the sum of squared residuals between model predictions and data. Returns the updated model parameters and the sum of squared residuals for each data batch.

Also restricts the version of numpy to < 2.0.0 because this otherwise leads to conflicts with keras.

Introduces a `dtype` argument to all models to explicitly enforce the dtype of the computations/parameters. All inputs and parameters are converted to this dtype. By default (i.e., when `dtype=None`), this is the default dtype in keras, `keras.config.floatx()`, which can be changed globally using `keras.config.set_floatx(dtype)`. Thus, users can both change the dtype globally and overwrite the global dtype locally with the `dtype` argument for specific models.

Adds a fix to the Gaussian pRF response function to normalize by the volume, so that it basically becomes the normal distribution probability density function. This was not done in the original pRF paper, but improves the parameter recovery of sigma and amplitudes.

### CODE REVIEW

File: src/prfmodel/fitters/linear.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        Fit a population receptive field model to target data.
+
+        Parameters
+        ----------
+        data : Tensor
+            Target data to fit the model to. Must have shape (num_batches, num_frames), where `num_batches` is the
+            number of batches for which parameters are estimated and `num_frames` is the number of time steps.
+        parameters : pandas.DataFrame
+            Dataframe with model parameters. Columns must contain different model parameters and
+            rows parameter values for each batch in `data`.
```
[the remark this answers is not in the document]
Comment (maltelueken) [reply src/prfmodel/fitters/linear.py]: If I change this here, I would need to change it in all the model functions. Perhaps we should also put this in a central place and just refer to it here. I will open a new issue for this (#50).

File: tests/fitters/test_sgd.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """Tests for SGDFitter class.
+
+    Uses a `Gaussian2DPRFModel` model with a `keras.optimizers.Adam` optimizer and `keras.losses.MeanSquaredError` loss
+    as a test case.
+
+    """
+
+    num_steps: int = 10
+
+    def _check_history(self, history: SGDHistory) -> None:
```
[the remark this answers is not in the document]
Comment (maltelueken) [reply tests/fitters/test_sgd.py]: I added helper methods to the tests for the other fitters too.
